prow/bugzilla/client.go

```diff
@@ -117,6 +119,105 @@ func (c *client) GetClones(bug *Bug) ([]*Bug, error) {
 	return getClones(c, bug)
 }
 
+// Gets children clones recursively using a mechanism similar to bfs
+func getRecursiveClones(c Client, root *Bug) ([]*Bug, error) {
+	var errs []error
+	var bug *Bug
+	clones := []*Bug{}
+	childrenQ := []*Bug{}
+	childrenQ = append(childrenQ, root)
+	// FYI Cannot think of any situation for circular clones
+	// But might need to revisit in case there are infinite loops at any point
+	for len(childrenQ) > 0 {
+		bug, childrenQ = childrenQ[0], childrenQ[1:]
+		clones = append(clones, bug)
+		children, err := getClones(c, bug)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("Error finding clones Bug#%d: %v", bug.ID, err))
+		}
+		if len(children) > 0 {
+			childrenQ = append(childrenQ, children...)
+		}
+	}
+	return clones, utilerrors.NewAggregate(errs)
+}
+
+// GetImmediateParents gets the Immediate parents of bugs with a matching summary
+func getImmediateParents(c Client, bug *Bug) ([]*Bug, error) {
+	var errs []error
+	parents := []*Bug{}
+	// One option would be to return as soon as the first parent is found
+	// ideally that should be enough, although there is a check in the getRoot function to verify this
+	// Logs would need to be monitored to verify this behavior
+	for _, parentID := range bug.DependsOn {
+		parent, err := c.GetBug(parentID)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("Failed to get parent bug #%d: %v", parentID, err))
+			continue
+		}
+		if parent.Summary == bug.Summary {
```

Comment:
Why do we need this check?

Reply:
According to my understanding the depends on field can contain parent bugs which are not clone parents, and only the ones with matching summaries are clone parents. Is that wrong?

Reply:
Hm I see, yeah it seems we do it like that elsewhere as well